logictestccl: TestReadCommittedLogic_select_for_share_write_buffering failed with unstable durability for lock key

pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test.TestReadCommittedLogic_select_for_share_write_buffering [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21297481?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21297481?buildTab=artifacts#/) on master @ [de52b5c9a99eedd416e39b2320289758891b93af](https://github.com/cockroachdb/cockroach/commits/de52b5c9a99eedd416e39b2320289758891b93af):


```
=== RUN   TestReadCommittedLogic_select_for_share_write_buffering
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/0824bbaf431fb104bed365fbfe465f16/logTestReadCommittedLogic_select_for_share_write_buffering2632983570
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    test_server_shim.go:150: cluster virtualization disabled due to issue: #156124 (expected label: C-bug)
[09:55:15] setting distsql_workmem='62849B';
    logictestbase.go:589: statement/query skipped with reason: unsupported configuration local-read-committed, statement/query only supports local-repeatable-read (no issue given)
    logictestbase.go:589: statement/query skipped with reason: unsupported configuration local-read-committed, statement/query only supports local-repeatable-read (no issue given)
    logic.go:3265: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19967/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test_/local-read-committed_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_share_write_buffering:149: SELECT database_name, schema_name, table_name, lock_key_pretty, lock_strength, durability, replace(isolation_level, ' ', '_') AS isolation_level, granted, contended FROM crdb_internal.cluster_locks
        expected:
            database_name  schema_name  table_name  lock_key_pretty   lock_strength  durability    isolation_level  granted  contended
            test           public       t           /Table/106/1/2/0  Shared         Unreplicated  READ_COMMITTED   true     false
            test           public       t           /Table/106/1/2/0  Shared         Unreplicated  READ_COMMITTED   true     false
            
        but found (query options: "colnames,retry,rowsort" -> ignore the following ordering of rows) :
            database_name  schema_name  table_name  lock_key_pretty   lock_strength  durability    isolation_level  granted  contended
            test           public       t           /Table/106/1/2/0  Shared         Unreplicated  READ_COMMITTED   true     false
            test           public       t           /Table/106/1/2/0  Shared         Replicated    READ_COMMITTED   true     false
            
[09:56:00] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19967/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test_/local-read-committed_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_share_write_buffering: 26 statements
[09:56:00] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19967/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test_/local-read-committed_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_share_write_buffering with config local-read-committed: 26 tests, 1 failures
    logic.go:4685: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19967/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test_/local-read-committed_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_share_write_buffering:156: error while processing
    logic.go:4685: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19967/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test_/local-read-committed_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_share_write_buffering:156: too many errors encountered, skipping the rest of the input
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0824bbaf431fb104bed365fbfe465f16/logTestReadCommittedLogic_select_for_share_write_buffering2632983570
--- FAIL: TestReadCommittedLogic_select_for_share_write_buffering (46.28s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReadCommittedLogic_select_for_share_write_buffering.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62476

--- a/pkg/sql/logictest/testdata/logic_test/select_for_share_write_buffering
+++ b/pkg/sql/logictest/testdata/logic_test/select_for_share_write_buffering
@@ -64,6 +64,11 @@ SET kv_transaction_buffered_writes_enabled = true;
 statement async writeReq count 1
 UPDATE t SET a = 2 WHERE a = 1
 
+# Since we executed the stmt above asynchronously, we're likely to get a new
+# connection here, so we must set the session variable on it too.
+statement ok
+SET kv_transaction_buffered_writes_enabled = true;
+
 onlyif config local-read-committed
 query TTTTTTTBB colnames,retry,rowsort
 SELECT database_name, schema_name, table_name, lock_key_pretty, lock_strength, durability, replace(isolation_level, ' ', '_') AS isolation_level, granted, contended FROM crdb_internal.cluster_locks

